### PR TITLE
Exchange object and file targets in the pipeline

### PR DESCRIPTION
**Pull request by algorsky**
I exchanged object and file targets in the pipeline. Specifically, I changed site_data_csv and site_info_csv into objects called site_data and site_info. I changed site_data_clean and cleaned_data into .csv files called nwis_data_clean_csv and cleaned_data_csv.

### CODE REVIEW

**Comment by algorsky** [comment]
> targets::tar_make()
✓ skip target site_data_01436690
✓ skip target site_data_01427207
✓ skip target site_data_01432160
✓ skip target site_data_01466500
✓ skip target site_data
✓ skip target nwis_data_clean_csv
✓ skip target site_info
• start target cleaned_data_csv
Rows: 140316 Columns: 3
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (1): site_no
dbl  (1): water_temperature
dttm (1): dateTime
\
ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
• built target cleaned_data_csv
✓ skip target figure_1_png
• end pipeline

File: 2_process/src/process_and_style.R [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
 
-annotate_and_style_data <- function(site_data_clean, site_filename){
-  site_info <- read_csv(site_filename)
-  annotated_data <- left_join(site_data_clean, site_info, by = "site_no") %>% 
+annotate_and_style_data <- function(filein, site_info, fileout){
+  site_data_csv<- read_csv(filein)
+  annotated_data_csv <- left_join(site_data_csv, site_info, by = "site_no") %>% 
     select(station_name = station_nm, site_no, dateTime, water_temperature, latitude = dec_lat_va, longitude = dec_long_va)%>%
     mutate(station_name = as.factor(station_name))
+    write_csv(annotated_data_csv, fileout)
```
Comment (lindsayplatt) [inline 2_process/src/process_and_style.R]: I think maybe there is a bit of confusion on when to use `_csv` in the naming. This should be used in the `_targets.R` files when the target you are creating is a file and so the output is a string of the filepath (when you do `tar_load([target_name])` the resulting object in your environment is something like `"out/my_file.csv"`). Internally to the function, I think your `filein` and `fileout` conventions are great. 

In both of these lines where you have `site_data_csv` and `annotated_data_csv` as objects, those objects actually contain the data - both are data.frames. By naming them with `_csv` here, you are implying that they contain the filepath but they contain the data! So, in both of these cases, you should drop `_csv`.

File: 2_process/src/process_and_style.R [hunk 2]
```diff
@@ -1,17 +1,17 @@
-process_data <- function(site_data_filename){
-  nwis_data<- read_csv(site_data_filename)
-  nwis_data_clean <- rename(nwis_data, water_temperature = X_00010_00000) %>% 
+process_data <- function(site_data, fileout){
+  nwis_data_clean_csv <- rename(site_data, water_temperature = X_00010_00000) %>%
```
Comment (lindsayplatt) [inline 2_process/src/process_and_style.R]: same idea here as I wrote below: `nwis_data_clean_csv` is a good name for a file _target_. Right now, you are creating an object with the data.frame of actual cleaned data. So you don't need the `_csv` appended to the object name.

File: _targets.R [hunk 3]
```diff
@@ -12,39 +12,38 @@ p1_targets_list <- list(
   tar_target(site_data_01436690, download_nwis_site_data('01436690')),
   tar_target(site_data_01466500, download_nwis_site_data('01466500')),
   
-  tar_target(site_data_csv,
+  tar_target(site_data,
```
Comment (lindsayplatt) [inline _targets.R:15]: Nice changes to this object!

File: _targets.R [hunk 4]
```diff
@@ -12,39 +12,38 @@ p1_targets_list <- list(
...
+    site_info,
+    nwis_site_info(site_data = site_data)
   )
 )
 
 p2_targets_list <- list(
   tar_target(
-    site_data_clean, 
-    process_data(site_data_csv)
+    nwis_data_clean_csv,
```
Comment (lindsayplatt) [inline _targets.R:32]: Yes, great target-naming here!

File: _targets.R [hunk 5]
```diff
@@ -12,39 +12,38 @@ p1_targets_list <- list(
...
   )
 )
 
 p2_targets_list <- list(
   tar_target(
-    site_data_clean, 
-    process_data(site_data_csv)
+    nwis_data_clean_csv, 
+    process_data(site_data = site_data, 
+                 fileout = "2_process/out/nwis_data_clean.csv")
```
Comment (lindsayplatt) [inline _targets.R]: Just need to add `format = "file"` for this to truly become a file target.

File: _targets.R [hunk 6]
```diff
@@ -12,39 +12,38 @@ p1_targets_list <- list(
...
   ),
   tar_target(
-    cleaned_data,
-    annotate_and_style_data(site_data_clean, site_filename = site_info_csv)
+    cleaned_data_csv,
```
Comment (lindsayplatt) [inline _targets.R]: Yes! Great target-naming 🎉

File: _targets.R [hunk 7]
```diff
@@ -12,39 +12,38 @@ p1_targets_list <- list(
...
+    annotate_and_style_data(filein = nwis_data_clean_csv, site_info = site_info,
+                            fileout = "2_process/out/cleaned_data.csv")
```
Comment (lindsayplatt) [inline _targets.R]: Just need to add `format = "file"` here so that this target will be tracked by `targets` as a file.

**Review by lindsayplatt — CHANGES_REQUESTED** [review]
This is great practice for following the file target naming conventions we discussed!

**Pull request merged**